builder/dockerfile/builder.go

```diff
@@ -316,6 +316,10 @@ func (b *Builder) hasFromImage() bool {
 //
 // TODO: Remove?
 func BuildFromConfig(config *container.Config, changes []string) (*container.Config, error) {
+	if len(changes) == 0 {
+		return config, nil
+	}
```

Comment:
This change wasn't strictly necessary, but I think it helps clarify the behaviour of this function when changes is empty.

`BuildFromConfig()` needs a much larger cleanup, but this PR isn't the place.